### PR TITLE
Improve overrides exporter take2

### PR DESCRIPTION
**Pull request by gouthamve**

### CODE REVIEW

File: pkg/commands/overrides_exporter.go [hunk 1]
```diff
@@ -27,19 +27,27 @@ type OverridesExporterCommand struct {
...
-			Help: "Various different limits.",
-		}, []string{"limit_type", "type", "user"}),
+		presetsGauge: promauto.With(registry).NewGaugeVec(prometheus.GaugeOpts{
+			Name: "cortex_overrides_presets",
+			Help: "Preset limits.",
+		}, []string{"limit_type", "name"}),
+		lastLimits: map[string]*validation.Limits{},
 	}
+
+	registry.Register(validation.NewOverridesExporter(oc))
```
Comment (pstibrany) [inline pkg/commands/overrides_exporter.go:48]: We need to wrap registry with `type="tenant"` label, and register OverridesExporter into it.

Comment (gouthamve) [reply pkg/commands/overrides_exporter.go:48]: I removed that. I moved presets to their own metrics which I think is cleaner imo.

Comment (pstibrany) [reply pkg/commands/overrides_exporter.go:48]: Oh, sorry, I see that presets now use different metric name.

**Review by pstibrany — APPROVED** [review]
LGTM, with caveat that there are changes that need to be reflected in our production.

* `cortex_overrides{type="preset"}` is now `cortex_overrides_presets`
* `cortex_overrides_presets` uses `limit_type` as before, but `cortex_overrides` now uses `limit_name` (I'd suggest renaming `limit_type` to `limit_name` for `cortex_overrides_presets` too)

**Comment by gouthamve** [comment]
Lets get this merged before we proceed here: https://github.com/grafana/cortex-tools/pull/155/

**Review by pstibrany — APPROVED** [review]
LGTM

**Pull request merged**